autoscaling.tf

```
@@ -1,9 +1,19 @@
+locals {
+  autoscaled_table = (
+    var.create_table && var.autoscaling_enabled ?
+    var.ignore_changes_global_secondary_index ?
+    aws_dynamodb_table.autoscaled[0] :
+    aws_dynamodb_table.autoscaled_gsi_ignore[0] :
+    ""
+  )
+}
+
 resource "aws_appautoscaling_target" "table_read" {
   count = var.create_table && var.autoscaling_enabled && length(var.autoscaling_read) > 0 ? 1 : 0
 
   max_capacity       = var.autoscaling_read["max_capacity"]
   min_capacity       = var.read_capacity
-  resource_id        = "table/${aws_dynamodb_table.autoscaled[0].name}"
+  resource_id        = "table/${local.autoscaled_table.name}"
```

```suggestion
  resource_id        = "table/${try(aws_dynamodb_table.autoscaled[0].name, aws_dynamodb_table.autoscaled_gsi_ignore[0].name)}"
```

There is no need to introduce a conditional local value `autoscaled_table`.

---

Isn't it simpler to have a variable rather than repeating the try everywhere though?